pkg/oauth/external/handler.go

```diff
@@ -211,12 +211,14 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 }
 
 func (h *Handler) handleError(err error, w http.ResponseWriter, req *http.Request) {
-	handled, err := h.errorHandler.AuthenticationError(err, w, req)
+	handled, _ := h.errorHandler.AuthenticationError(err, w, req)
 	if handled {
 		return
 	}
+
+	klog.V(4).Infof("handle error failed for err: %v", err)
 	w.WriteHeader(http.StatusInternalServerError)
-	w.Write([]byte(`An error occurred`))
+	http.Error(w, "An error occured", http.StatusInternalServerError)
```

Comment:
You'll want to remove the `w.WriteHeader` above - https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/net/http/server.go;l=2093